packages/components/src/components/wizard/wizard.jsx

```diff
@@ -1,86 +1,97 @@
+        }
+    }, [initial_step, getSteps]);
+
+    const onChangeStep = stats => {
+        // User callback
+        onStepChange(stats);
     };
 
-    const prevStep = () => {
-        const prev_step = step - 1;
-        const is_first_step = prev_step < 0;
-        if (is_first_step) return;
-        setStep(prev_step);
-        setComponent(steps[prev_step].component);
+    const isInvalidStep = next => next < 0 || next >= getTotalSteps();
+
+    const onSetActiveStep = next => {
+        if (active_step === next || isInvalidStep(next)) return;
+        setActiveStep(next);
+        onChangeStep({
+            active_step: next + 1,
+        });
     };
 
-    const selectStep = selected_step => {
-        const is_prev_step_not_allowed = selected_step < 0;
-        const is_next_step_not_allowed = selected_step > last_step;
-        if (is_prev_step_not_allowed || is_next_step_not_allowed) return;
-        setStep(selected_step);
-        setComponent(steps[selected_step].component);
+    const getSteps = React.useCallback(() => React.Children.toArray(children), [children]);
+
+    const getCurrentStep = () => active_step + 1;
+
+    const getTotalSteps = () => getSteps().length;
+
+    const goToStep = step => onSetActiveStep(step - 1);
+
+    const goToFirstStep = () => goToStep(1);
+
+    const goToLastStep = () => goToStep(getTotalSteps());
+
+    const goToNextStep = () => onSetActiveStep(active_step + 1);
+
+    const goToPreviousStep = () => onSetActiveStep(active_step - 1);
+
+    // Allows for using HTML elements as a step
+    const isReactComponent = ({ type }) => typeof type === 'function' || typeof type === 'object';
+
+    const properties = {
+        getCurrentStep,
+        getTotalSteps,
+        goToNextStep,
+        goToPreviousStep,
+        goToStep,
+        goToFirstStep,
+        goToLastStep,
     };
 
-    React.useImperativeHandle(ref, () => ({
-        prevStep,
-        nextStep,
-        selectStep,
-    }));
-
-    const prevButton = () =>
-        step === 0 ? (
-            <div className='wizard__placeholder' />
-        ) : (
-            <div role='button' onClick={prevStep} className='wizard__btn wizard__btn-left'>
-                <Icon icon='IcArrowLeftBold' className='wizard__btn-icon-left' />
-                {lbl_previous}
-            </div>
-        );
-
-    const nextButton = () =>
-        step === last_step ? (
-            <div className='wizard__placeholder' />
-        ) : (
-            <div role='button' onClick={nextStep} className='wizard__btn wizard__btn-right'>
-                {lbl_next}
-                <Icon icon='IcArrowRightBold' className='wizard__btn-icon-right' />
-            </div>
-        );
+    const childrenWithProps = React.Children.map(getSteps(), (child, i) => {
+        if (!child) return null;
+
+        if (i === active_step)
+            return <Wizard.Step>{isReactComponent(child) ? React.cloneElement(child, properties) : child}</Wizard.Step>;
```

Comment:
child can only be a React component right? Do we need an explicit check here?

Reply:
@balakrishna-binary children can be both React component or html elements, on line 62 `React.Children` can even pass null or undefined, hence isReactComponent makes sure to clone react element in case of passing array or html element

Reply:
Got it 👍